mewarpx/examples/thermionic_diode_rz_cylinder.py

```diff
@@ -0,0 +1,194 @@
+import argparse
+import sys
+
+from mewarpx.utils_store import util as mwxutil
+
+mwxutil.init_libwarpx(ndim=2, rz=True)
+
+import numpy as np
+from pywarpx import picmi
+
+from mewarpx import assemblies, diags, emission, mespecies, runinfo
+from mewarpx.mwxrun import mwxrun
+
+
+class VacuumTEC_Cylinder(object):
+
+    #######################################################################
+    # Begin global user parameters                                        #
+    #######################################################################
+
+    CATHODE_TEMP = 1100 + 273.15 # K
+    CATHODE_A = 6e5              # A/m^2/K^2
+    CATHODE_PHI = 2.11           # eV
+    USE_SCHOTTKY = False
+    V_CATHODE = -CATHODE_PHI     # cathode is grounded
+
+    ANODE_TEMP = 200             # K
+    ANODE_PHI = 1.4              # eV
+
+    D_CA = 50e-6                 # m
+
+    NZ = 32
+    NR = 512
+
+    DT = 0.5e-12                 # s
+
+    NPPC = 4
+
+    #######################################################################
+    # End global user parameters and user input                           #
+    #######################################################################
+
+    def setup_run(self):
+
+        self.rmax = 1e-3
+        self.rmin = self.rmax - self.D_CA
+        self.zmax = (self.rmax - self.rmin) / self.NR * self.NZ / 2.0
+        self.zmin = -self.zmax
+
+        self.V_ANODE = self.V_CATHODE + self.V_ANODE_CATHODE
+
+        self.DIAG_STEPS = (
+            (self.TOTAL_TIMESTEPS // 5) if self.TOTAL_TIMESTEPS > 10
+            else self.TOTAL_TIMESTEPS
+        )
+
+        #######################################################################
+        # Set geometry, boundary conditions and timestep                      #
+        #######################################################################
+
+        mwxrun.init_grid(
+            lower_bound=[self.rmin, self.zmin],
+            upper_bound=[self.rmax, self.zmax],
+            number_of_cells=[self.NR, self.NZ], min_tiles=1,
+            bc_fields_z_min='periodic', bc_fields_z_max='periodic',
+            bc_particles_z_min='periodic', bc_particles_z_max='periodic',
+            bc_fields_r_min='dirichlet', bc_fields_r_max='dirichlet',
+        )
+        mwxrun.init_timestep(DT=self.DT)
+        mwxrun.simulation.max_steps = self.TOTAL_TIMESTEPS
+        mwxrun.simulation.load_balance_intervals = self.TOTAL_TIMESTEPS // 5000
```

Comment:
What happens if this evaluates to 0? Does that just disable load balancing? (I honestly don't know, just want to make sure it's not an error to set it to 0.)

Reply:
Yeap, the default value is 0 so in that case load balancing just doesn't occur.